src/main/java/teammates/common/util/Config.java

```diff
@@ -103,53 +98,113 @@ public final class Config {
         APP_MAINTAINERS = Collections.unmodifiableList(
-                Arrays.asList(properties.getProperty("app.maintainers", "").split(",")));
-        SUPPORT_EMAIL = properties.getProperty("app.crashreport.email");
-        EMAIL_SENDEREMAIL = properties.getProperty("app.email.senderemail");
-        EMAIL_SENDERNAME = properties.getProperty("app.email.sendername");
-        EMAIL_REPLYTO = properties.getProperty("app.email.replyto");
-        EMAIL_SERVICE = properties.getProperty("app.email.service");
-        SENDGRID_APIKEY = properties.getProperty("app.sendgrid.apikey");
-        MAILGUN_APIKEY = properties.getProperty("app.mailgun.apikey");
-        MAILGUN_DOMAINNAME = properties.getProperty("app.mailgun.domainname");
-        MAILJET_APIKEY = properties.getProperty("app.mailjet.apikey");
-        MAILJET_SECRETKEY = properties.getProperty("app.mailjet.secretkey");
-        SEARCH_SERVICE_HOST = properties.getProperty("app.search.service.host");
-        ENABLE_DATASTORE_BACKUP = Boolean.parseBoolean(properties.getProperty("app.enable.datastore.backup", "false"));
-        MAINTENANCE = Boolean.parseBoolean(properties.getProperty("app.maintenance", "false"));
+                Arrays.asList(getProperty(properties, devProperties, "app.maintainers", "").split(",")));
+        SUPPORT_EMAIL = getProperty(properties, devProperties, "app.crashreport.email");
+        EMAIL_SENDEREMAIL = getProperty(properties, devProperties, "app.email.senderemail");
+        EMAIL_SENDERNAME = getProperty(properties, devProperties, "app.email.sendername");
+        EMAIL_REPLYTO = getProperty(properties, devProperties, "app.email.replyto");
+        EMAIL_SERVICE = getProperty(properties, devProperties, "app.email.service");
+        SENDGRID_APIKEY = getProperty(properties, devProperties, "app.sendgrid.apikey");
+        MAILGUN_APIKEY = getProperty(properties, devProperties, "app.mailgun.apikey");
+        MAILGUN_DOMAINNAME = getProperty(properties, devProperties, "app.mailgun.domainname");
+        MAILJET_APIKEY = getProperty(properties, devProperties, "app.mailjet.apikey");
+        MAILJET_SECRETKEY = getProperty(properties, devProperties, "app.mailjet.secretkey");
+        SEARCH_SERVICE_HOST = getProperty(properties, devProperties, "app.search.service.host");
+        ENABLE_DATASTORE_BACKUP = Boolean.parseBoolean(
+                getProperty(properties, devProperties, "app.enable.datastore.backup", "false"));
+        MAINTENANCE = Boolean.parseBoolean(getProperty(properties, devProperties, "app.maintenance", "false"));
+
+        // Dev property values, read directly from devProperties
+        APP_LOCALDATASTORE_PORT = Integer.parseInt(
+                devProperties.getProperty("app.localdatastore.port", "8484"));
+        ENABLE_DEVSERVER_LOGIN = Boolean.parseBoolean(
+                devProperties.getProperty("app.enable.devserver.login", "true"));
```

Comment:
This particular property should now default to `false`. Although "only relevant in dev server", this one is actually used even for production server (see `isDevServerLoginEnabled()`), and in the event that `IS_DEV_SERVER` stops working properly, this is the last resort that stops the dev server login page from being displayed to public.